src/main/java/org/ehrbase/fhirbridge/ehr/converter/TherapyCompositionConverter.java

```diff
@@ -0,0 +1,264 @@
+                // Map Medizingeraet for RESP
+                Coding usedCodeCoding = procedure.getUsedCode().get(0).getCoding().get(0);
+
+                if (usedCodeCoding.getSystem().equals(SNOMED_SYSTEM) && geraetenameMap.containsKey(usedCodeCoding.getCode())) {
+
+                    MedizingeraetCluster medizingeraetCluster = new MedizingeraetCluster();
+
+                    medizingeraetCluster.setGeraetenameDefiningCode(geraetenameMap.get(usedCodeCoding.getCode()));
+
+                    durchgefuehrteProzedur.setMedizingeraet(new ArrayList<>());
+                    durchgefuehrteProzedur.getMedizingeraet().add(medizingeraetCluster);
+                } else {
+                    throw new UnprocessableEntityException("Invalid medical device code");
+                }
+            }
+
+            durchgefuehrteProzedur.setArtDerProzedurDefiningCode(composition.getKategorie().get(0).getValue());
+
+
+            if (!procedure.getExtension().isEmpty()) {
+                for (Extension extension : procedure.getExtension()) {
+                    if (extension.getValue() instanceof Coding) {
+                        durchgefuehrteProzedur.setDurchfuehrungsabsichtValue(((Coding) extension.getValue()).getDisplay());
+                        break;
+                    }
+                }
+
+            }
+
+            durchgefuehrteProzedur.setKommentarValue(procedure.getNote().toString());
+
+            if (procedure.getPerformedDateTimeType() != null && procedure.getPerformedDateTimeType().getValueAsCalendar() != null) {
+                durchgefuehrteProzedur.setTimeValue(procedure.getPerformedDateTimeType().getValueAsCalendar().toZonedDateTime());
+            } else {
+                durchgefuehrteProzedur.setTimeValue(composition.getStartTimeValue());
+            }
+
+
+        } catch (Exception e) {
+            throw new CompositionConversionException("Some parts of the present procedure did not contain the required elements. "
+                    + e.getMessage(), e);
+        }
+
+        durchgefuehrteProzedur.setLanguage(Language.DE);
+        durchgefuehrteProzedur.setSubject(new PartySelf());
+        //durchgefuehrteProzedur.setCareflowStepDefiningCode(CareflowStepDefiningCode.GEPLANTE_PROZEDUR);
+        durchgefuehrteProzedur.setCurrentStateDefiningCode(CurrentStateDefiningCode.PLANNED);
+        composition.setProzedur(durchgefuehrteProzedur);
+    }
+
+    private void mapNotDone(Procedure procedure, GECCOProzedurComposition composition) {
+
+        NichtDurchgefuehrteProzedurEvaluation nichtDurchgefuehrteProzedur = new NichtDurchgefuehrteProzedurEvaluation();
+
+        // TODO: Check whether this has to be an enum type
+        nichtDurchgefuehrteProzedur.setAussageUeberDenAusschlussValue(procedure.getStatus().getDisplay());
+        try {
+            nichtDurchgefuehrteProzedur.setEingriffDefiningCode(mapNameDerProzedur(procedure));
+        } catch (Exception e) {
```

Comment:
Do you really want to catch ANY exception here or a specific one ?

Reply:
Fixed